xbmc/utils/Signals.h

```diff
@@ -0,0 +1,201 @@
+/*
```

Comment:
Moved for now as suggested. ActorProtocol can really do anything can't it :-) fwiw the signal/slot approach here is similar to what most toolkits like Gtk+ and Qt do and basically a (very) bare-bones version of Boost.Signals2.